ui/src/app/machines/views/MachineDetails/MachinePCIDevices/MachinePCIDevices.tsx

```diff
@@ -1,89 +1,24 @@
 import { useEffect } from "react";
 
-import { Button, Col, Icon, Row, Strip } from "@canonical/react-components";
-import pluralize from "pluralize";
 import { useDispatch, useSelector } from "react-redux";
 import { useParams } from "react-router";
-import { Link } from "react-router-dom";
 
 import type { SetSelectedAction } from "../MachineSummary";
+import NodeDevices from "../NodeDevices";
 
-import DoubleRow from "app/base/components/DoubleRow";
-import Placeholder from "app/base/components/Placeholder";
-import { HardwareType } from "app/base/enum";
 import { useWindowTitle } from "app/base/hooks";
 import type { RouteParams } from "app/base/types";
 import machineSelectors from "app/store/machine/selectors";
-import type { Machine } from "app/store/machine/types";
 import { actions as nodeDeviceActions } from "app/store/nodedevice";
 import nodeDeviceSelectors from "app/store/nodedevice/selectors";
 import { NodeDeviceBus } from "app/store/nodedevice/types";
-import type { NodeDevice } from "app/store/nodedevice/types";
 import type { RootState } from "app/store/root/types";
-import { NodeActions } from "app/store/types/node";
-
-type NodeDeviceGroup = {
-  hardwareTypes: HardwareType[];
-  items: NodeDevice[];
-  label: string;
-  pathname?: string;
-};
 
 type Props = { setSelectedAction: SetSelectedAction };
 
-const generateGroup = (group: NodeDeviceGroup, machine: Machine) =>
-  group.items.map((nodeDevice, i) => {
-    const {
-      commissioning_driver,
-      id,
-      numa_node_id,
-      pci_address,
-      product_id,
-      product_name,
-      vendor_id,
-      vendor_name,
-    } = nodeDevice;
-    const numaNode =
-      "numa_nodes" in machine
-        ? machine.numa_nodes.find((numa) => numa.id === numa_node_id)
-        : null;
-
-    return (
-      <tr key={`pci-device-${id}`}>
-        <td>
-          {i === 0 && (
-            <DoubleRow
-              data-test="group-label"
-              primary={
-                <strong>
-                  {group.pathname ? (
-                    <Link to={`/machine/${machine.system_id}${group.pathname}`}>
-                      {group.label}
-                    </Link>
-                  ) : (
-                    group.label
-                  )}
-                </strong>
-              }
-              secondary={pluralize("device", group.items.length, true)}
-            />
-          )}
-        </td>
-        <td>
-          <DoubleRow primary={vendor_name} secondary={vendor_id} />
-        </td>
-        <td>{product_name}</td>
-        <td>{product_id}</td>
-        <td>{commissioning_driver}</td>
-        <td className="u-align--right" data-test="pci-numa">
-          {numaNode?.index}
-        </td>
-        <td className="u-align--right">{pci_address}</td>
-      </tr>
-    );
-  });
-
-const MachinePCIDevices = ({ setSelectedAction }: Props): JSX.Element => {
+const MachinePCIDevices = ({
```

Comment:
It looks like the only differences remaining between this component and `MachineUSBDevices` is the window title and setting `bus={NodeDeviceBus.PCIE}`. Are these wrapping components necessary anymore? Could all the selectors etc. be moved into `NodeDevices` and just set the bus and window title, or possibly remove the wrapping components and set the bus in [MachineDetails.tsx](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/machines/views/MachineDetails/MachineDetails.tsx)?

Reply:
Yeah, originally I had it like you said but I felt like it was a level of abstraction too far. For sure it reduces the amount of repeated code, but I like the idea of treating the PCI tab and USB tabs as separate wrapping components, if only because they're separate tabs in the UI.

I did get caught up in the idea that it would be better if `NodeDevices` was a "dumb" component, but thinking about it again I don't think that actually helps at all. I'll move the redux code into `NodeDevices` but I think I'll still keep the wrapping components.

Reply:
OK, that seems reasonable.